internal/cache/cache_test.go

```diff
@@ -345,21 +347,23 @@ func TestCanAuthenticate(t *testing.T) {
 	t.Parallel()
 
 	tests := map[string]struct {
-		userPasswords  map[string]string
-		useoldaccounts bool
-		shadowMode     *int
+		userPasswords       map[string]string
+		useoldaccounts      bool
+		shadowMode          *int
+		disabledOfflineAuth bool
 
 		wantErr bool
 	}{
 		"can authenticate one user":                     {userPasswords: map[string]string{"myuser@domain.com": "my password"}},
 		"handle separately multiple users and password": {userPasswords: map[string]string{"myuser@domain.com": "my password", "otheruser@domain.com": "other password"}},
 		"can authenticate even with shadow file RO":     {userPasswords: map[string]string{"myuser@domain.com": "my password"}, shadowMode: &cache.ShadowROMode},
+		"can authenticate even with very old user":      {userPasswords: map[string]string{"veryolduser@domain.com": "my password"}, useoldaccounts: true},
 
 		// error cases
 		"error on wrong password":                         {userPasswords: map[string]string{"myuser@domain.com": "wrong password"}, wantErr: true},
 		"error on wrong user":                             {userPasswords: map[string]string{"does not exist user": "my password"}, wantErr: true},
 		"error on checking when can’t access shadow file": {userPasswords: map[string]string{"myuser@domain.com": "my password"}, shadowMode: &cache.ShadowNotAvailableMode, wantErr: true},
-		"do not let too old unpurged accounts to log in ": {userPasswords: map[string]string{"veryolduser@domain.com": "my password"}, useoldaccounts: true, wantErr: true},
```

Comment:
We don’t have any test left for old unpurged account not being able to log in (meaning, accounts with last login time between 2*expiration and expiration)?

Reply:
This test is still removed, no?

Reply:
I agree that "error on offline authentication disabled" is going to the same code path, but here, we are using with an user which shouldn’t be purged with the default policy. I think it’s good to test here with an user which should have been purged too (so an additional test).

Reply:
Indeed, the test is missing. But it's because we can no longer test it, I guess. If the credentials expiration is set to 0, the user is supposed to be able to authenticate despite how long he re-validated online... So we either have a case in which the user is purged (which is the case "error on offline with purged user") or a case in which credentials do not expire (which is the case "offline, connect purged user from cache").